src/util/stream.rs

```diff
@@ -22,6 +23,63 @@ use std::time::Duration;
 ///
 /// [`timeout`]: #method.timeout
 pub trait StreamExt: Stream {
+    /// Debounce the stream on the trailing edge using the given duration.
+    ///
+    /// Errors will pass through without being debounced. Debouncing will
+    /// happen on the trailing edge. This means all items (except the last
+    /// one) will be discarded until the delay has elapsed without an item
+    /// being passed through. The last item that was passed through will
+    /// be returned.
+    ///
+    /// Care must be taken that this stream returns `Async::NotReady` at some point,
+    /// otherwise the debouncing implementation will overflow the stack during
+    /// `.poll()` (i. e. don't use this directly on `stream::repeat`).
+    ///
+    /// See also [`debounce_builder`], which allows more configuration over how the
+    /// debouncing is done.
+    ///
+    /// [`debounce_builder`]: #method.debounce_builder
+    fn debounce(self, dur: Duration) -> Debounce<Self>
+    where Self: Sized
+    {
+        self.debounce_builder()
+            .duration(dur)
+            .edge(Edge::Trailing)
+            .build()
+    }
+
+    /// Create a builder that builds a debounced version of this stream.
+    ///
+    /// The returned builder can be used to configure the debouncing process.
+    ///
+    /// Care must be taken that this stream returns `Async::NotReady` at some point,
+    /// otherwise the debouncing implementation will overflow the stack during
+    /// `.poll()` (i. e. don't use this directly on `stream::repeat`).
+    fn debounce_builder(self) -> DebounceBuilder<Self>
```

Comment:
This probably shouldn't be a trait fn. The builder can be constructed directly with `Builder::new`.